src/sign.ml

```diff
@@ -277,6 +282,29 @@ let add_rule : t -> sym -> rule -> unit = fun sign sym r ->
     let m = (sym.sym_name, r) :: m in
     sign.sign_deps := PathMap.add sym.sym_path m !(sign.sign_deps)
 
+(** [add_rules s r] adds a bunch of rules [r] to its attached symbol to the
+    signature [s] and builds decision trees. *)
+let add_rules : t -> (sym * pp_hint * rule loc) list -> unit = fun sign rs ->
+  let add_rule (s, h, r) =
+    out 3 "(rule) %a\n" Print.pp_rule (s,h,r.elt) ;
+    add_rule sign s r.elt in
+  List.iter add_rule rs ;
+  (* EXPE *)
+  let build_tree symb =
+    match symb.sym_mode with
+    | Defin
+    | Injec when !(symb.sym_rules) <> [] ->
```

Comment:
What is this when for??

Reply:
Don't build a tree if there is no rule

Reply:
Oh I see. But it cannot happen because build_tree is called only on symbols having rules, isn't it?

Reply:
Ah indeed.